components/ledger/ledger-utxo-token-cache/src/main/kotlin/net/corda/ledger/utxo/token/cache/services/ClaimStateStore.kt

```diff
@@ -0,0 +1,8 @@
+package net.corda.ledger.utxo.token.cache.services
+
+import net.corda.data.ledger.utxo.token.selection.state.TokenPoolCacheState
+import java.util.concurrent.CompletableFuture
+
+interface ClaimStateStore {
+    fun enqueueRequest(request: (TokenPoolCacheState) -> TokenPoolCacheState):CompletableFuture<Boolean>
```

Comment:
```suggestion
    fun enqueueRequest(request: (TokenPoolCacheState) -> TokenPoolCacheState): CompletableFuture<Boolean>
```
Also needs KDoc explaining what `Boolean` wrapped into the `CompletableFuture` is meant to represent.